moviesImporter/scripts/importer.py

```diff
@@ -1,53 +1,49 @@
 import csv
 import sys
 from moviesImporter.models import Movie
-from IPython import embed
+from datetime import datetime
+
 
 def run():
+    startTime = datetime.now()
     csv.field_size_limit(sys.maxsize)
-    with open('moviesImporter/data_to_import/title.basics.tsv') as tsvfile:
-        basics = csv.DictReader(tsvfile, dialect='excel-tab')
-        with open('moviesImporter/data_to_import/title.crew.tsv') as tsvfile:
-            crew_members = list(csv.DictReader(tsvfile, dialect='excel-tab'))
-            
-            for title in basics:
-                if title['titleType'] == 'movie' and title['genres'] and title['startYear']:
-                    found_index = None
-                    for index, crew in enumerate(crew_members):
-                        if crew['tconst'] == title['tconst'] and crew['directors'] != '\\N':
-                            try: 
-                                found_index = index
-                                movie_dict = create_movie_dict(crew, title)
-                                Movie(**movie_dict).save()
-                                break 
-                            except Exception as e:
-                                print(e)
-                                print(f"{title['primaryTitle']} did not save.") 
+    with open('moviesImporter/data_to_import/title.basics.tsv', 'r') as basics_file:
+        basics = csv.DictReader(basics_file, dialect='excel-tab')
+        with open('moviesImporter/data_to_import/title.crew.tsv', 'r') as crew_file:
+            movies_to_save = []
+            crew_file_array = list(csv.reader(crew_file, dialect='excel-tab'))
+            c = 0
+            for index, title in enumerate(basics):
+                if title['titleType'] == 'movie' and title['genres'] and title['genres'] != '\\N' and title['startYear'] and title['startYear'] != '\\N':
+                    crew_line = crew_file_array[index + 1]
```

Comment:
The solution was much simpler than I thought.

Because both files were the exact same length, I started thinking maybe they had the same IDs in the same order 🤔

So, I ran a small test of about 100k records and found that to be the case 🎉

That makes our problem an order of magnitude faster, as now we just loop through the titles and use the index of each title to select the corresponding crew line 💸